docs/architecture/adr-038-state-listening.md

```diff
@@ -472,18 +735,34 @@ func NewSimApp(
 		}
 	}
 
+=======
+>>>>>>> Stashed changes
 	...
 
+	// enable streaming
+	enableKey := fmt.Sprintf("%s.%s", baseapp.StreamingTomlKey, baseapp.StreamingEnableTomlKey)
+	if enable := cast.ToBool(appOpts.Get(enableKey)); enable {
+		pluginKey := fmt.Sprintf("%s.%s", baseapp.StreamingTomlKey, baseapp.StreamingPluginTomlKey)
+		pluginName := cast.ToString(appOpts.Get(pluginKey))
+		logLevel := cast.ToString(appOpts.Get(flags.FlagLogLevel))
+		plugin, err := streaming.NewStreamingPlugin(pluginName, logLevel)
+		if err != nil {
+			tmos.Exit(err.Error())
+		}
+		if err := baseapp.RegisterStreamingService(bApp, appOpts, appCodec, keys, plugin); err != nil {
+			tmos.Exit(err.Error())
+		}
+	}
+
```

Comment:
nit: would be nice to abstract this into BaseApp so users don't have to set this. Many times users will forget. We (sdk) team can tackle this post merge though